src/libraries/UniswapV4Lib.sol

```diff
@@ -190,11 +184,9 @@ library UniswapV4Lib {
             uint8(Actions.TAKE_ALL)
         );
 
-        bool zeroForOne = tokenIn == Currency.unwrap(poolKey.currency0);
-
-        address tokenOut = zeroForOne
-            ? Currency.unwrap(poolKey.currency1)
-            : Currency.unwrap(poolKey.currency0);
+        ( address tokenIn, address tokenOut ) = zeroForOne
+            ? (Currency.unwrap(poolKey.currency0), Currency.unwrap(poolKey.currency1))
+            : (Currency.unwrap(poolKey.currency1), Currency.unwrap(poolKey.currency0));
```

Comment:
Order of operations should be:

require(slippage != 0)
PoolKey =
(tokenIn, tokenOut)
require(_getNormalizedBalance)
triggerRateLimitDecrease
actions = 
params =